sparkplug-core/src/clojure/sparkplug/core.clj

```diff
@@ -261,20 +263,18 @@
   "Construct an RDD representing the cartesian product of two RDDs. Returns a
   new pair RDD containing all combinations of elements between the datasets."
   ^JavaPairRDD
-  [^JavaRDD rdd1 ^JavaRDD rdd2]
+  [^JavaRDDLike rdd1 ^JavaRDDLike rdd2]
   (rdd/set-callsite-name
     (.cartesian rdd1 rdd2)))
 
 
 (defn union
   "Construct a union of the elements in the provided RDDs. Any identical
   elements will appear multiple times."
-  ^JavaRDD
-  ([^JavaRDD rdd1 ^JavaRDD rdd2]
+  ([rdd1 rdd2]
    (rdd/set-callsite-name
      (.union rdd1 rdd2)))
-  ^JavaRDD
-  ([^JavaRDD rdd1 ^JavaRDD rdd2 & rdds]
+  ([rdd1 rdd2 & rdds]
```

Comment:
Guessing this was the spot that led to the protocol/varargs questions?

Reply:
Yep :)